src/main/template/vpc.yaml

```diff
@@ -0,0 +1,410 @@
+        - Fn::Cidr:
+          - !Ref cidrBlock
+          - "4"
+          - !Ref firstQuarteringPartitionBits
+      MapPublicIpOnLaunch: false
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} Private Subnet AZ C
+
+  publicSubnetA:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref vpc
+      MapPublicIpOnLaunch: true
+      AvailabilityZone: !Select
+        - 0
+        - Fn::GetAZs: ""
+      CidrBlock: !Select
+        - 0
+        - Fn::Cidr:
+          - !Select
+            - 3
+            - Fn::Cidr:
+              - !Ref cidrBlock
+              - "4"
+              - !Ref firstQuarteringPartitionBits
+          - "4"
+          - !Ref secondQuarteringPartitionBits
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} Public Subnet AZ A
+
+  publicSubnetB:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref vpc
+      MapPublicIpOnLaunch: true
+      AvailabilityZone: !Select
+        - 1
+        - Fn::GetAZs: ""
+      CidrBlock: !Select
+        - 1
+        - Fn::Cidr:
+          - !Select
+            - 3
+            - Fn::Cidr:
+              - !Ref cidrBlock
+              - "4"
+              - !Ref firstQuarteringPartitionBits
+          - "4"
+          - !Ref secondQuarteringPartitionBits
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} Public Subnet AZ B
+
+  publicSubnetC:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref vpc
+      MapPublicIpOnLaunch: true
+      AvailabilityZone: !Select
+        - 2
+        - Fn::GetAZs: ""
+      CidrBlock: !Select
+        - 2
+        - Fn::Cidr:
+          - !Select
+            - 3
+            - Fn::Cidr:
+              - !Ref cidrBlock
+              - "4"
+              - !Ref firstQuarteringPartitionBits
+          - "4"
+          - !Ref secondQuarteringPartitionBits
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} Public Subnet AZ C
+
+  publicRouteTable:
+    Type: AWS::EC2::RouteTable
+    Properties:
+      VpcId: !Ref vpc
+      Tags:
+        - Key: Name
+          Value: !Sub ${AWS::StackName} Public Route Table
+
+  publicRoute:
+    Type: AWS::EC2::Route
+    Properties:
+      RouteTableId: !Ref publicRouteTable
+      DestinationCidrBlock: 0.0.0.0/0
+      GatewayId: !Ref internetGateway
+
+  publicSubnetARouteTableAssociation:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref publicRouteTable
+      SubnetId: !Ref publicSubnetA
+
+  publicSubnetBRouteTableAssociation:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref publicRouteTable
+      SubnetId: !Ref publicSubnetB
+
+  publicSubnetCRouteTableAssociation:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref publicRouteTable
+      SubnetId: !Ref publicSubnetC
+
+  natGatewayAElasticIP:
+    Type: AWS::EC2::EIP
+    Properties:
+      Domain: vpc
```

Comment:
Because this is a race condition. If this elastic IP were to be created before the VPC gateway attachment, it would fail. In the current order of things, this does not happen. But by luck, not by enforcement.